From 4c93c52361080e49494116fde78fa46d182afff2 Mon Sep 17 00:00:00 2001
From: deepongi <infectedmushi@gmail.com>
Date: Sat, 6 Dec 2025 18:44:00 +0000
Subject: [PATCH] fix LKM spoof patch to use CONFIG_KSU

---
 kernel/supercalls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/supercalls.c b/kernel/supercalls.c
index 805896f0..3706d0b0 100644
--- a/kernel/supercalls.c
+++ b/kernel/supercalls.c
@@ -78,7 +78,7 @@ static int do_get_info(void __user *arg)
 {
     struct ksu_get_info_cmd cmd = {.version = KERNEL_SU_VERSION, .flags = 0};
 
-#ifdef MODULE
+#if defined(MODULE) || defined(CONFIG_KSU)
     cmd.flags |= 0x1;
 #endif
 
-- 
2.52.0

